sdktests/server_side_stream_request.go

```diff
@@ -0,0 +1,52 @@
+package sdktests
+
+import (
+	"strings"
+	"time"
+
+	"github.com/launchdarkly/sdk-test-harness/framework/ldtest"
+	"github.com/launchdarkly/sdk-test-harness/mockld"
+	"github.com/launchdarkly/sdk-test-harness/servicedef"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func doServerSideStreamRequestTests(t *ldtest.T) {
+	t.Run("headers", func(t *ldtest.T) {
+		sdkKey := "my-sdk-key"
+
+		dataSource := NewSDKDataSource(t, mockld.EmptyServerSDKData())
+		_ = NewSDKClient(t, WithConfig(servicedef.SDKConfigParams{
+			Credential: sdkKey,
+		}), dataSource)
+
+		request := expectRequest(t, dataSource.Endpoint(), time.Second)
+
+		assert.Equal(t, sdkKey, request.Headers.Get("Authorization"))
+		assert.NotEqual(t, sdkKey, request.Headers.Get("User-Agent"))
+	})
+
+	t.Run("URL path is correct when base URI has a trailing slash", func(t *ldtest.T) {
```

Comment:
We've been burned by this before in several SDKs, due to overly simple-minded path concatenation logic.